src/cartridge/src/cartridge.cpp

```diff
@@ -25,7 +24,15 @@ Cartridge::Cartridge(std::string rom_path) {
 	rom_file.seekg(0);
 
 	// Read the file
-	auto byte_input = std::vector<char>(rom_file_size, '\0');
+	auto byte_input = std::vector<char>();
+	try{
+		byte_input = std::vector<char>(rom_file_size, '\0');
```

Comment:
The line that needs to be in the try catch is not this line, it's the place where the file is actually being opened. 

```cpp
	auto rom_file = std::ifstream(rom_path, std::ifstream::in | std::ios::binary);
```

But I would argue that it's probably better to enclose the entire method in a try-catch, because the file reading might go wrong too.

Reply:
So this is what I had in mind. I opened the file after accepting the ROM path and checked if the file opened correctly. If yes, then the `main.cpp` proceeded to create a `unique ptr` to the `Cartridge` Object. Here I put a try catch block in case the given path was a Directory. It would throw an exception of `bad alloc` here. 
But I agree with you, enclosing the entire constructor in a try-catch block is better and might avoid confusions in the future.